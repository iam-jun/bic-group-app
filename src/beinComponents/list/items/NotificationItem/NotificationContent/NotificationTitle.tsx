import React from 'react';
import {StyleSheet} from 'react-native';
import i18n from 'i18next';

import Text from '~/beinComponents/Text';

export interface NotificationTitleProps {
  testID?: string;
  actorNames: string;
  verbText?: any;
  groupText?: string;
}

const NotificationTitle = ({
  testID,
  actorNames,
  verbText,
  groupText,
}: NotificationTitleProps) => {
  const styles = createStyle();

  let _verbText = verbText;
  if (typeof _verbText === 'object') {
    const {emoji, targetText} = verbText;

    _verbText = (
      <React.Fragment>
        {i18n.t('reacted') + ' '}
        {emoji}
        {' ' + targetText}
      </React.Fragment>
    );
  }

  return (
    <Text.BodyM testID={testID || 'notification_title'} style={styles.title}>
      {actorNames + ' '}
      <Text.Body testID="notification_title.verb_text">
        {_verbText}
        {groupText && (
          <Text.BodyM testID="notification_title.group_text">
            {' ' + groupText}
          </Text.BodyM>
        )}
      </Text.Body>
    </Text.BodyM>
  );
};

const createStyle = () => {
  return StyleSheet.create({
    title: {
      alignItems: 'baseline',
    },
  });
};

export default NotificationTitle;
