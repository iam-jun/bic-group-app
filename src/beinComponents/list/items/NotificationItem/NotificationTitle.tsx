import React from 'react';
import {StyleSheet} from 'react-native';
import Text from '~/beinComponents/Text';
import i18n from '~/localization';
interface Props {
  actorNames: string;
  verbText?: any;
  groupText?: string;
}

const NotificationTitle = ({actorNames, verbText, groupText}: Props) => {
  const styles = createStyle();

  const renderReactVerb = () => {
    if (!verbText) return null;

    if (typeof verbText === 'string') return verbText;

    const {emoji, targetText} = verbText;

    return (
      <React.Fragment>
        {i18n.t('reacted') + ' '}
        {emoji}
        {' ' + targetText}
      </React.Fragment>
    );
  };

  return (
    <Text.BodyM style={styles.title}>
      {actorNames + ' '}
      <Text.Body>
        {renderReactVerb}
        {groupText && <Text.BodyM>{' ' + groupText}</Text.BodyM>}
      </Text.Body>
    </Text.BodyM>
  );
};

const createStyle = () => {
  return StyleSheet.create({
    title: {
      alignItems: 'baseline',
    },
  });
};

export default NotificationTitle;
