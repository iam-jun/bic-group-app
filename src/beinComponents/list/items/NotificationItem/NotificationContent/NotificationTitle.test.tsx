import {cleanup, render} from '@testing-library/react-native';
import React from 'react';
import NotificationTitle from './NotificationTitle';

afterEach(cleanup);

describe('NotificationTitle component', () => {
  const baseProps = {
    actorNames: 'Tráº§n Nam Anh',
    verbText: 'commented on your post',
  };

  it(`renders correctly`, async () => {
    const wrapper = render(<NotificationTitle {...baseProps} />);
    const rendered = wrapper.toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it(`should show "NotificationTitle" with groupText`, async () => {
    const props = {
      ...baseProps,
      groupText: 'EVOL Community',
    };
    const wrapper = render(<NotificationTitle {...props} />);
    const component = wrapper.getByTestId('notification_title.group_text');

    expect(component).not.toBeNull();
  });

  it(`should show "NotificationTitle" with emoji`, async () => {
    const props = {
      ...baseProps,
      verbText: {
        emoji: 'ðŸ˜„',
        targetText: 'to your post',
      },
    };
    const wrapper = render(<NotificationTitle {...props} />);
    const rendered = wrapper.toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
