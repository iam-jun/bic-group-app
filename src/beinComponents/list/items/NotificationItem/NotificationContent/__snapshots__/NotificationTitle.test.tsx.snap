// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationTitle component renders correctly 1`] = `
<Text
  style={
    Object {
      "alignItems": "baseline",
      "color": "#2F3945",
      "fontFamily": "OpenSans-SemiBold",
      "fontSize": 16,
    }
  }
  testID="notification_title"
>
  Tráº§n Nam Anh 
  <Text
    style={
      Object {
        "color": "#2F3945",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      }
    }
    testID="notification_title.verb_text"
  >
    commented on your post
  </Text>
</Text>
`;

exports[`NotificationTitle component should show "NotificationTitle" with emoji 1`] = `
<Text
  style={
    Object {
      "alignItems": "baseline",
      "color": "#2F3945",
      "fontFamily": "OpenSans-SemiBold",
      "fontSize": 16,
    }
  }
  testID="notification_title"
>
  Tráº§n Nam Anh 
  <Text
    style={
      Object {
        "color": "#2F3945",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      }
    }
    testID="notification_title.verb_text"
  >
    undefined 
    ðŸ˜„
     to your post
  </Text>
</Text>
`;
